### DISCUSSION TITLE
Release 2.12.0. NEW: Auto-scalable. Font selector by language. Languages are dynamic loaded with the po/mo lang files.  XWayland fully compatible. Added  Russian, French, German, Spanish, Portuguese, Italian, Dutch, Greek, Chinese, Indonesian, Hebrew, Arabic and Esperanto layout languages.    Added Zeos mse componentes. And much more...

### DISCUSSION THREAD

**Discussion by fredvs** [fredvs/ideU#66] [Announcements]
ideU the ide. Universal.

Multi OS, multi languages, multi compilers, voice assisted.

Included MSEgui form designer
https://github.com/mse-org/mseide-msegui
plugin fpGUI uidesigner_ext
https://github.com/fredvs/uidesigner_ext
and sak voice assisted kit.
https://github.com/fredvs/sak

For Windows 64/32, Linux 64/32, FreeBSD 64 and Raspberry Pi ARM.
Added release for Raspberry Pi ARM-aarch64.

ideU is based on MSEide by Martin Schreiber.
https://github.com/mse-org/mseide-msegui

Many thanks to Johann Elsass and Sieghard for their excellent ideas, bug fixes and kind support.

NEW: For Windows and Linux, **ideU_PacPas** is a complete Pascal workshop including _ideU_ IDE, _fpc-ootb_ compiler and (for Windows) _gdb_ debugger.  Included source of _msegui_, _fpgui_ and _bgrabitmap_. (see assets).

In new release was added:

-  Lot of fixes.
-  Auto-scalable  with screen resolution used.
-  Languages are dynamic loaded with the po/mo lang files and font selector by language.
-  Fully XWayland and fpc 3.3.1 compatible.
-  Compiled with last msegui source and fpc 3.2.2.
-  Russian, Ukrainian, French, German, Spanish, Portuguese, Italian, Romanian, Dutch, Greek, Chinese, Indonesian, Hebrew, Arabic and Esperanto layout languages.  
-  Zeos mse components.
-  And much more...

Silver:
![ideU2862](https://user-images.githubusercontent.com/3421249/196552302-fa752645-5c12-4eef-aea0-fd5140fe9ca7.png)

Carbon:
![ideU286dark](https://user-images.githubusercontent.com/3421249/196552391-5ca82673-8089-42d6-87ad-095cd4431d10.png)

Here how to install ideU:

- If you upgrade ideU, download the release, unzip it and re-initialize the macros with menu | Setting | General Setting | and click "OK".

- If you are new with ideU:
   
   Note that all the files that follow are included in the **_PacPas_** release.    

   If you want to compile Pascal source, you need a installed Free Pascal compiler.
   You can download the FPC compiler here: http://www.freepascal.org/download.var
   or/and the FPC-OOTB compiler there: https://github.com/fredvs/freepascal-ootb/releases
     
   If you want to compile MSEgui application, you need to download MSEgui source code.
   https://github.com/mse-org/mseide-msegui

   If you want to compile fpGUI application, you need to download fpGUI source code.
   https://github.com/graemeg/fpGUI/tree/maint

   If you want to compile BGRAbitmap application, you need to download BGRAbitmap source code.
   https://github.com/bgrabitmap/bgrabitmap/releases
    
  Then download ideU release binary (see assets).
  Unzip it somewhere, launch ideU(.exe) and set up all the paths.

  If you want to compile-use last commit of ideU, then download ideU source code.
  https://github.com/fredvs/ideU/archive/refs/heads/main.zip
  And compile the project-file: /ideu/src/ideU.prj

Compiling ideU from commandline in directory ideu/src/:

for Linux:

`fpc -oideU -Fu/directory_of_mseide-msegui/lib/common/kernel/linux -Fu/directory_of_mseide-msegui/lib/common/* -Mobjfpc -Sh -Fcutf8 -dmse_with_zeoslib -gl -B -Xs -CX -XX -O1 -dmse_dynpo -SIcorba -FUunits -vewinhq -dusemo ideU.pas
`

for Windows:

`fpc.exe -oideU.exe -Fu/directory_of_mseide-msegui/lib/common/kernel/windows -Fu/directory_of_mseide-msegui/lib/common/* -Mobjfpc -Sh -Fcutf8 -dmse_with_zeoslib -B -Xs -CX -XX -O1 -WG -dmse_dynpo -SIcorba -FUunits -vewinhq -dusemo ideU.pas
`

  When ideU is re-compiled, copy the binary in /src/ideU(.exe) and replace the old binary of ideU(.exe) that is in release directory with the copy of the new one.

  This because in release is added all the plugins that are not in the source.

Have lot of fun! 

Fre;D

<hr /><em>This discussion was created from the release <a href='https://github.com/fredvs/ideU/releases/tag/2.12.0'>Release 2.10.0. NEW: Auto-scalable. Font selector by language. Languages are dynamic loaded with the po/mo lang files.  XWayland fully compatible. Added  Russian, French, German, Spanish, Portuguese, Italian, Dutch, Greek, Chinese, Indonesian, Hebrew, Arabic and Esperanto layout languages.    Added Zeos mse componentes. And much more...</a>.</em>

**Comment by GrahamNZ** [comment]
Hi Fred,

One of the additions you mention is "Auto-scalable with screen resolution used". Do you mean the ide auto scales now, or do you mean it is possible to create programs that auto scale? In Lazarus I can use some project options combined with _Application.Scaled:=True;_ but I'm not sure if there is an equivalent in ideU with MSEgui.

Thanks.

**Comment by fredvs** [comment]
It is about ideU.
The auto-scalable feature of ideU was done by code.

I used the same code for https://github.com/fredvs/strumpract.
(I have to check where is the code for autoscalable)
